### DISCUSSION TITLE
what is the best way to encrypt sensitive data?

### DISCUSSION THREAD

**Discussion by ruslandanilin** [vectordotdev/vector#12537] [Q&A] [answered]
Hi Guys!

Assume that there is a field with some sensitive data in your log message and using [redact](https://vector.dev/docs/reference/vrl/functions/#redact) is not an option because you need to read it later. 

What is the best way to encrypt data with Vector? Are there any solutions out of the box?
Thank you

**Comment by spencergilbert** [comment, marked as answer]
It's not been released to stable yet, but we've recently added [`encrypt`](https://master.vector.dev/docs/reference/vrl/functions/#encrypt) and [`decrypt`](https://master.vector.dev/docs/reference/vrl/functions/#decrypt) functions which should be available in nightly releases.

**Reply by ruslandanilin** [reply]
Thank you @spencergilbert! This is exactly what we need.
